common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -899,14 +899,17 @@ public RefreshTokenRecord getFamilyRefreshTokenForHomeAccountId(@NonNull final S
                                                            @NonNull AccountRecord accountRecord) {
         final List<IdTokenRecord> result = new ArrayList<>();
 
+        final List<Credential> allCredentials = mAccountCredentialCache.getCredentials();
```

Comment:
```suggestion
        // Load all the credentials to inspect once, such that we don't need to requery the cache
        // pass these into the new getCredentialsFilteredBy overload, rather than hit disk again
        final List<Credential> allCredentials = mAccountCredentialCache.getCredentials();
```